Dockerfile.builder

```diff
@@ -21,23 +21,18 @@ RUN apt-get update && \
     apt-get upgrade -y --no-install-recommends && \
     apt-get install -y --no-install-recommends \
       # Base Cilium-build dependencies
-      apt-utils \
       binutils \
       coreutils \
       curl \
       gcc \
       git \
       libc6-dev \
       libelf-dev \
-      m4 \
       make \
-      pkg-config \
-      python \
       rsync \
       unzip \
       wget \
```

Comment:
Could we replace `wget` with `curl` where it's needed and get rid of it?

Reply:
Makes sense, the rsync/unzip/curl/wget/zip combo is not quite clear to me yet where it's needed, I'll see to reduce it further.